src/picomsg/src/routing_table/RoutingTable.hpp

```diff
@@ -0,0 +1,86 @@
+
+#pragma once
+
+#include "routing_table/c_api/routing_table.h"
+#include "utils/functors.hpp"
+#include <mutex>
+#include <unordered_map>
+#include <memory>
+#include <vector>
+#include <functional>
+
+namespace picomsg {
+namespace detail {
+
+class RouteSPKey {
+private:
+	struct routing_table_route_entry_sp_info m_route_entry_sp_info;
+
+public:
+	explicit RouteSPKey(const struct routing_table_route_entry_sp_info& route_entry_sp_info);
+
+	friend bool operator==(const RouteSPKey& lhs, const RouteSPKey& rhs) noexcept {
+		return pm_ipsec_ts_equals(&lhs.m_route_entry_sp_info.traffic_selector, &rhs.m_route_entry_sp_info.traffic_selector) &&
+				(lhs.m_route_entry_sp_info.vlan_id == rhs.m_route_entry_sp_info.vlan_id);
+	}
+
+	friend struct std::hash<RouteSPKey>;
```

Comment:
I think befriending `std::hash` is redundant here since you've defined the hashing functor in `std` namespace.